3.8/release-notes-upgrading-changes38.md

```diff
@@ -130,6 +126,49 @@ However, to restore the pre-3.8 behavior, it is possible to set the value of
 this option to `1`. The value can even be set to `0` to disable using the
 scheduler's queue fill grade as an (un)availability indicator.
 
+
+#### AQL query memory limits
+
+ArangoDB 3.8 introduces a default memory limit for AQL queries to prevent rogue queries 
+from consuming the entire memory available to an arangod instance.
+
+The memory limit is introduced via changing the default value of the startup option 
+`--query.memory-limit` from previously `0` (meaning: no limit) to a dynamically calculated 
+value. The per-query memory limits defaults are now:
+
+```
+Available memory:    134217728    (128MiB)  Limit:            0      (0MiB), %mem:  0.0
+Available memory:    268435456    (256MiB)  Limit:            0      (0MiB), %mem:  0.0
```

Comment:
Not sure how to interpret these two. Is there no limit or is the limit 0 bytes, i.e. no queries can be run at all? `%mem: 0.0` suggests that no memory is available for queries.

Reply:
A limit of `0` internally means "no limit". This is also what users can set explicitly.
The reason for this is that we make a basic assumption in our formular that at least 256 MB should be put aside for other things (e.g. operating system), and that we only start caring for larger memory volumes.
But obviously all of this is not rocket science and is based on some assumptions that may or may not be true.

We can change the formula so that it makes sense for us, e.g.:
```
MEM: 33554432 (32MB): 8388608 (8MB), %RAM: 25
MEM: 67108864 (64MB): 16777216 (16MB), %RAM: 25
MEM: 134217728 (128MB): 33554432 (32MB), %RAM: 25
MEM: 201326592 (192MB): 50331648 (48MB), %RAM: 25
MEM: 268435456 (256MB): 67108864 (64MB), %RAM: 25
MEM: 536870912 (512MB): 201326592 (192MB), %RAM: 37.5
MEM: 805306368 (768MB): 402653184 (384MB), %RAM: 50
MEM: 1073741824 (1024MB): 603979776 (576MB), %RAM: 56.25
MEM: 2147483648 (2048MB): 1288490189 (1228MB), %RAM: 60
MEM: 4294967296 (4096MB): 2576980377 (2457MB), %RAM: 60
MEM: 8589934592 (8192MB): 5153960755 (4915MB), %RAM: 60
MEM: 17179869184 (16384MB): 10307921511 (9830MB), %RAM: 60
MEM: 25769803776 (24576MB): 15461882265 (14745MB), %RAM: 60
MEM: 34359738368 (32768MB): 20615843021 (19660MB), %RAM: 60
MEM: 42949672960 (40960MB): 25769803776 (24576MB), %RAM: 60
MEM: 68719476736 (65536MB): 41231686041 (39321MB), %RAM: 60
MEM: 103079215104 (98304MB): 61847529063 (58982MB), %RAM: 60
MEM: 137438953472 (131072MB): 82463372083 (78643MB), %RAM: 60
MEM: 274877906944 (262144MB): 164926744167 (157286MB), %RAM: 60
MEM: 549755813888 (524288MB): 329853488333 (314572MB), %RAM: 60
```